curtsies/formatstring.py

```diff
@@ -239,30 +282,53 @@ def request(self, max_width):
 
                 # if not adding it us makes us short, this must have been a double-width character
                 if width < max_width:
-                    assert width + 1 == max_width, 'unicode character width of more than 2!?!'
-                    assert w == 2, 'unicode character of width other than 2?'
-                    return (width + 1, Chunk(s[start_offset:self.internal_offset] + replacement_char,
-                                             atts=self.chunk.atts))
-                return (width, Chunk(s[start_offset:self.internal_offset], atts=self.chunk.atts))
+                    assert (
```

Comment:
This is so much worse, ignore this part. A lot of this code would have been written differently if we were using Black back then, but unless we refactor it I think we shoudl add ignores.